src/uproot/behaviors/TH1.py

```diff
@@ -21,7 +21,15 @@ def _boost_axis(axis, metadata):
     fXbins = axis.member("fXbins", none_if_missing=True)
 
     if axis.member("fLabels") is not None:
-        out = boost_histogram.axis.StrCategory([str(x) for x in axis.member("fLabels")])
+        try:
+            out = boost_histogram.axis.IntCategory(
+                [int(x) for x in axis.member("fLabels")]
+            )
+        except ValueError:
+            # not all labels can be interpreted as integers
+            out = boost_histogram.axis.StrCategory(
+                [str(x) for x in axis.member("fLabels")]
+            )
```

Comment:
Try-except logic is dangerous in that it can hide more exceptions than the one you want. We don't know whether `axis.member("fLabels")` raised the ValueError, `int(x)` raised it, or `IntCategory`. They have to be more focused:

```suggestion
        fLabels = axis.member("fLabels")
        try:
            labels = [int(x) for x in fLabels]
            category_cls = boost_histogram.axis.IntCategory
        except ValueError:
            labels = [str(x) for x in fLabels]
            category_cls = boost_histogram.axis.StrCategory
        out = category_cls(labels)
```